src/Adapter/Order/OrderPresenter.php

```diff
@@ -0,0 +1,255 @@
+<?php
+
+namespace PrestaShop\PrestaShop\Adapter\Order;
+
+use PrestaShop\PrestaShop\Adapter\Cart\CartPresenter;
+use PrestaShop\PrestaShop\Adapter\ObjectSerializer;
+use PrestaShop\PrestaShop\Adapter\Product\PricePresenter;
+use PrestaShop\PrestaShop\Adapter\Translator;
+use PrestaShop\PrestaShop\Adapter\LegacyContext;
+use Address;
+use AddressFormat;
+use Carrier;
+use Cart;
+use Context;
+use CustomerMessage;
+use HistoryController;
+use Order;
+use Tools;
+
+class OrderPresenter
+{
+    /* @var ObjectSerializer */
+    private $objectSerializer;
+    /* @var PricePresenter */
+    private $pricePresenter;
+    /* @var Translator */
+    private $translator;
+
+    public function __construct()
+    {
+        $this->objectSerializer = new ObjectSerializer();
+        $this->pricePresenter = new PricePresenter();
+        $this->translator = new Translator(new LegacyContext());
+    }
+
+    /**
+     * @param Order $order
+     *
+     * @return array
+     */
+    public function present(Order $order)
+    {
+        return [
+            'products' => $this->getProducts($order),
+            'amounts' => $this->getAmounts($order),
```

Comment:
The `CartPresenter` returns this at the moment:

![image](https://cloud.githubusercontent.com/assets/1460499/13218959/62ff4020-d96d-11e5-8990-c4af277f9b20.png)

And I think we should try to make them converge so that a template designed for a cart works with an order and conversely.

Since `CartPresenter` is more used than `OrderPresenter` (which is new), I think the path of least effort is to make `OrderPresenter` closer to `CartPresenter` and:

- move  `total` and `subtotals` out of `amounts` and put them at the root of the returned array
- add `products_count`
- add a `summary_string` (e.g. "N" products for a total of "X" CURRENCY)
- add `id_address_delivery` and `id_address_invoice`

This way, `OrderPresenter` has the same behaviour as a `CartPresenter`, only it adds info and thus a cart template can be used safely with an order and order templates could inherit (and extend) cart templates.